avaframe/com1DFAPy/com1DFAtools.py

```diff
@@ -162,33 +168,51 @@ def intializeResistance(dem):
 
 
 def DFAIterate(cfg, particles, fields, dem, Ment, Cres, Tcpu):
+    """ Perform computations and save results for desired interval """
+
+    # Load configuration settings
     Tend = cfg.getfloat('Tend')
     dtSave = cfg.getfloat('dtSave')
-    dt = cfg.getfloat('dt')
 
+    # Initialise Lists to save fields
     Particles = [copy.deepcopy(particles)]
     Fields = [copy.deepcopy(fields)]
     # save Z, S, U, T at each time step for developping purpouses
     Z = np.empty((0, 0))
     S = np.empty((0, 0))
     U = np.empty((0, 0))
     T = np.empty((0, 0))
-    # initialize time and counters
+
+    # Initialize time and counters
     nSave = 1
     nIter = 0
     iterate = True
     particles['iterate'] = iterate
     t = particles['t']
     # Start time step computation
     while t < Tend and iterate:
+
+        #++++++++++++++++if you want to use cfl time step+++++++++++++++++++
+        # CALL TIME STEP:
+        # to play around with the courant number
+        cfg['cMax'] = '0.5'
+        # dtStable = tD.getcflTimeStep(particles, dem, cfg)
+        # dt overwrites dt in .ini file, so comment this block if you dont want to use cfl
+        #++++++++++++++++++++++++++++++++++++++++++++++
+        # get time step
+        dt = cfg.getfloat('dt')
         t = t + dt
         nIter = nIter + 1
         log.debug('Computing time step t = %f s', t)
         T = np.append(T, t)
         particles['t'] = t
         Tcpu['nSave'] = nSave
 
-        particles, fields, Tcpu = computeTimeStep(cfg, particles, fields, dem, Ment, Cres, Tcpu)
+        # Perform computations
+        # choose which time stepping
+        # particles, fields, Tcpu, dt = computeLeapFrogTimeStep(cfg, particles, fields, dt, dem, Ment, Cres, Tcpu)
+        particles, fields, Tcpu = computeTimeStep(cfg, particles, fields, dt, dem, Ment, Cres, Tcpu)
```

Comment:
switch this with a feature flag instead of commenting it

Reply:
done